File: solved/1424.py
from typing import List


class Solution:
    def findDiagonalOrder(self, nums: List[List[int]]) -> List[int]:
        offset = []
        for y in range(0, len(nums)):
            for x in range(0, len(nums)):
                if(len(offset) <= (x+y)):
                    offset.append([nums[y][x]])
                else:
                    offset[y+x].append(nums[y][x])
        toReturn = []
        for l in offset:
            for i in range(len(l), 0, -1):
                toReturn.append(i)
        return toReturn

        

File: solved/1424.dart
//import 'dart:math';

void main() {
  Solution().main();
}

class Solution {
  void main() {
    print(findDiagonalOrder([
      [1, 2, 3],
      [4, 5, 6],
      [7, 8, 9]
    ]));
  }

  List<int> findDiagonalOrder(List<List<int>> nums) {
    List<List<int>> offset = [];

    for (int y = 0; y < nums.length; y++) {
      for (int x = 0; x < nums[y].length; x++) {
        if (offset.length <= (x + y)) {
          offset.add([nums[y][x]]);
        } else {
          offset[y + x].add(nums[y][x]);
        }
      }
    }
    List<int> toReturn = [];
    for (List<int> row in offset) {
      for (int i = row.length - 1; i >= 0; i--) {
        toReturn.add(row[i]);
      }
    }
    return toReturn;
  }
}
